src/gdpr/index.ts

```diff
@@ -0,0 +1,68 @@
+/*
+Full API doc
+https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md
+
+How can scripts on a page determine if there is a CMP present?
+https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#how-can-scripts-on-a-page-determine-if-there-is-a-cmp-present
+
+Ping
+https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#ping
+*/
+
+interface PingResponse {
+  gdprApplies?: boolean;
+}
+
+export interface TCData {
+  gdprApplies?: boolean;
+  vendor: {
+    consents: { [vendorId: number]: boolean | undefined };
+  };
+}
+
+declare global {
+  interface Window {
+    __tcfapi(
+      command: 'ping',
+      version: number,
+      cb: (response: PingResponse) => void
+    ): void;
+
+    __tcfapi(
+      command: 'getTCData',
+      version: number,
+      cb: (tcData: TCData, success: boolean) => void
+    ): void;
+  }
+}
+
+/*
+Note: the api assumes that `undefined` is a still pending response, so hasGdprConsent would be false
+https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#what-does-the-gdprapplies-value-mean
+*/
+const hasConsent = (gdprApplies?: boolean): boolean => {
+  return gdprApplies === undefined ? false : gdprApplies;
+};
+
+export const hasGdprConsent = (vendorIds: number[]): Promise<boolean> => {
+  return new Promise((resolve) => {
+    // If the api is missing then assume no consent
+    if (window.__tcfapi === undefined) resolve(false);
```

Comment:
What if this is called before the API is available?

Should this not hook into something?

Reply:
Afaict this shouldn't be an issue from my reading of the API,

API implementors should provide a stub script initiailly before any other scripts that depend on it are loaded:
https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#how-does-the-cmp-stub-api-work

And scripts that rely on the API shouldn't need to check if it's loaded, scripts build up a queue and then execute them once fully loaded:
https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#how-can-scripts-determine-if-the-cmp-script-is-loaded-yet

Reply:
Yeah I thought there was a queue thing which needs to pop our function on and then we can wait till it get executed - but would that work here:

`if (window.__tcfapi === undefined) resolve(false);` this does not hook into a queue.

Or are you saying that the implementor should have stubbed that function earlier in the page then loading our script?

This seems error prone?